### ISSUE TITLE
GPU issue for nvidia 1060

### ISSUE THREAD

**Issue by rakuri255** [rakuri255/UltraSinger#17]
Hi again, I've tried for some reason is not detecting my Nvidia 1060 6 GB, according to this page https://developer.nvidia.com/cuda-gpus my GPU has CUDA 6.1, and according to this one https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel-22-09.html torch should be compatible with mine...

> GPU Requirements

> Release 22.09 supports CUDA compute capability 6.0 and later. This corresponds to GPUs in the NVIDIA Pascal, NVIDIA Volta™, NVIDIA Turing™, NVIDIA Ampere architecture, and NVIDIA Hopper™ architecture families. For a list of GPUs to which this compute capability corresponds, see [CUDA GPUs](https://developer.nvidia.com/cuda-gpus). For additional support details, see [Deep Learning Frameworks Support Matrix](https://docs.nvidia.com/deeplearning/frameworks/support-matrix/index.html).

I also installed CUDA driver following the readme, and executed this command: pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu117

_Originally posted by @BanCrash in https://github.com/rakuri255/UltraSinger/issues/15#issuecomment-1556156925_

**Comment by rakuri255** [comment]
> You also installed the new requirements from the .txt?

Yes, I did, but now that you mention it, I got this error:

![imagen](https://github.com/rakuri255/UltraSinger/assets/11770760/b28d29a9-af32-4b8b-829f-5520bce7958c)

I don't know how to resolve it, to be honest.

_Originally posted by @BanCrash in https://github.com/rakuri255/UltraSinger/issues/15#issuecomment-1556164038_

@BanCrash Maybe it is fixed with comit b8ca18dee7b1c92eb87462ede287e1d323a6d6ec
I loosen the tensorflow and removed the tensorflow-directml-plugin

**Comment by BanCrash** [comment]
Ok, now the requirements are fixed, but still not detecting the gpu:

![imagen](https://github.com/rakuri255/UltraSinger/assets/11770760/08206712-37a8-4849-84c8-ecc78204fa8b)

**Comment by rakuri255** [comment]
@BanCrash ok it seems that pip dont always want to grap the cuda `+cu117` package.

Please check if you have torch or torch+cu117 installed. 

Try this than to force the cu117 installation:
`pip3 install torch==2.0.1+cu117 torchvision==0.15.2+cu117 torchaudio==2.0.2+cu117 --index-url https://download.pytorch.org/whl/cu117`

**Comment by BanCrash** [comment]
> @BanCrash ok it seems that pip dont always want to grap the cuda `+cu117` package.
> 
> Please check if you have `torch` or `torch+cu117` installed.
> 
> Try this than to force the cu117 installation: `pip3 install torch==2.0.1+cu117 torchvision==0.15.2+cu117 torchaudio==2.0.2+cu117 --index-url https://download.pytorch.org/whl/cu117`

![imagen](https://github.com/rakuri255/UltraSinger/assets/11770760/ce470494-8452-464d-9aa6-f138489df262)

That fixed the issue, thank you!!

**Comment by rakuri255** [comment]
cant reproduce this. Please check the tensorflow packages and maybe reinstall these? 

This is what i have:
![grafik](https://github.com/rakuri255/UltraSinger/assets/36919102/50ac8274-5b6e-4c62-83e3-ceccc7ac16a6)

**Comment by BanCrash** [comment]
> cant reproduce this. Please check the tensorflow packages and maybe reinstall these?
> 
> This is what i have: ![grafik](https://user-images.githubusercontent.com/36919102/240025218-50ac8274-5b6e-4c62-83e3-ceccc7ac16a6.png)

I have the same as you, but after reinstalling them the issue is solved and the program now finished as expected. Thank you again!

**Comment by BanCrash** [comment]
> Hope that helps you now with your temperature ;) You can force whisper cpu usage if you need a more accurate model. See ReadMe
> 
> Will finish the full GPU support when i have time.

Yes, I think this will let me use it more than before. Thank you again!

**Comment by kubalakowski** [comment]
Isn't it related to TensorFlow > 2.10.0 not supporting GPU on Windows?

https://github.com/tensorflow/tensorflow/releases/tag/v2.11.1

> Note: TensorFlow 2.10 was the last TensorFlow release that supported GPU on native-Windows. Starting with TensorFlow 2.11, you will need to install TensorFlow in WSL2, or install tensorflow-cpu and, optionally, try the TensorFlow-DirectML-Plugin.

**Comment by rakuri255** [comment]
> Isn't it related to TensorFlow > 2.10.0 not supporting GPU on Windows?
> 
> https://github.com/tensorflow/tensorflow/releases/tag/v2.11.1
> 
> > Note: TensorFlow 2.10 was the last TensorFlow release that supported GPU on native-Windows. Starting with TensorFlow 2.11, you will need to install TensorFlow in WSL2, or install tensorflow-cpu and, optionally, try the TensorFlow-DirectML-Plugin.

Yes. But demucs and whisper also works currently with just torch+cu. For crepe i dont know yet.  But TensorFlow-DirectML-Plugin would solve the issue anyway.
